File: CS/CustomTimeRulerSample/MyTimeRulerFormatStringService.cs
using DevExpress.XtraScheduler;
using DevExpress.XtraScheduler.Services;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DXSample
{
#region #myservice    
public class MyTimeRulerFormatStringService : TimeRulerFormatStringServiceWrapper
    {
        public MyTimeRulerFormatStringService(ITimeRulerFormatStringService service)
            : base(service)
        {
        }
        public override string GetHalfDayHourFormat(TimeRuler ruler)
        {
            return ruler.UseClientTimeZone ? "htt" : "HH:mm";
        }
        public override string GetHourFormat(TimeRuler ruler)
        {
            return ruler.UseClientTimeZone ? "htt" : "HH:mm";
        }
        public override string GetHourOnlyFormat(TimeRuler ruler)
        {
            return ruler.UseClientTimeZone ? "htt" : "HH";
        }
        public override string GetMinutesOnlyFormat(TimeRuler ruler)
        {
            return ruler.UseClientTimeZone ? "''" : "mm";
        }
        public override string GetTimeDesignatorOnlyFormat(TimeRuler ruler)
        {
            return ruler.UseClientTimeZone ? "''" : "mm";
        }
    }
#endregion #myservice
}

File: VB/CustomTimeRulerSample/MyTimeRulerFormatStringService.vb
Imports DevExpress.XtraScheduler
Imports DevExpress.XtraScheduler.Services

Namespace DXSample

'#Region "#myservice    "
    Public Class MyTimeRulerFormatStringService
        Inherits TimeRulerFormatStringServiceWrapper

        Public Sub New(ByVal service As ITimeRulerFormatStringService)
            MyBase.New(service)
        End Sub

        Public Overrides Function GetHalfDayHourFormat(ByVal ruler As TimeRuler) As String
            Return If(ruler.UseClientTimeZone, "htt", "HH:mm")
        End Function

        Public Overrides Function GetHourFormat(ByVal ruler As TimeRuler) As String
            Return If(ruler.UseClientTimeZone, "htt", "HH:mm")
        End Function

        Public Overrides Function GetHourOnlyFormat(ByVal ruler As TimeRuler) As String
            Return If(ruler.UseClientTimeZone, "htt", "HH")
        End Function

        Public Overrides Function GetMinutesOnlyFormat(ByVal ruler As TimeRuler) As String
            Return If(ruler.UseClientTimeZone, "''", "mm")
        End Function

        Public Overrides Function GetTimeDesignatorOnlyFormat(ByVal ruler As TimeRuler) As String
            Return If(ruler.UseClientTimeZone, "''", "mm")
        End Function
    End Class
'#End Region  ' #myservice
End Namespace
